src/components/molecules/Visualizer/Engine/Cesium/cameraLimiter.ts

```diff
@@ -0,0 +1,237 @@
+import {
+  Cartesian3,
+  Cartographic,
+  Color,
+  EllipsoidGeodesic,
+  PolylineDashMaterialProperty,
+  Math,
+  Camera as CesiumCamera,
+  Rectangle,
+} from "cesium";
+import type { Viewer as CesiumViewer } from "cesium";
+import { useEffect, useMemo, useState } from "react";
+import { CesiumComponentRef } from "resium";
+import { RefObject } from "use-callback-ref/dist/es5/types";
+
+import { Camera } from "@reearth/util/value";
+
+import type { SceneProperty } from "..";
+
+import { getCamera } from "./common";
+
+export function useCameraLimiter(
+  cesium: RefObject<CesiumComponentRef<CesiumViewer>>,
+  camera: Camera | undefined,
+  property: SceneProperty["cameraLimiter"] | undefined,
+) {
+  const geodsic = useMemo(():
+    | undefined
+    | { geodesicVertical: EllipsoidGeodesic; geodesicHorizontal: EllipsoidGeodesic } => {
+    const viewer = cesium.current?.cesiumElement;
+    if (!viewer || viewer.isDestroyed()) return undefined;
```

Comment:
Correct me if I am wrong, but it looks like every function and useEffect inside useCameraLimiter declares viewer and then returns undefined if there is no viewer or if the viewer is destroyed.
Can't you just do it once in the useCameraLimiter's scope at the beginning? Then, you could simplify the other checks!

Reply:
It would give me `tslint` errors, and I'll have to always use the viewer like this: `viewer?.xproperty` and Cesium functions require viewer to exist.
But there is one place where I don't need this check and I removed it from there :+1: